TerminalDocs/customize-settings/key-bindings.md

```diff
@@ -0,0 +1,458 @@
+---
+title: Windows Terminal Key Bindings
+description: Learn how to create custom key bindings for Windows Terminal.
+author: cinnamon-msft
+ms.author: cinnamon
+ms.date: 05/19/2020
+ms.topic: overview
+ms.service: terminal
+---
+
+# Custom key bindings in Windows Terminal
+
+You can create custom key bindings inside the Windows Terminal that give you control of how you interact with the Terminal using your keyboard.
+
+## Key binding properties
+
+Key bindings can be constructed using the following properties.
+
+### Command
+
+The command executed when the associated keys are pressed.
+
+**Property name:** `command`
+
+**Necessity:** Required
+
+**Accepts:** String
+
+### Keys
+
+Defines the key combinations used to call the command.
+
+**Property name:** `keys`
+
+**Necessity:** Required
+
+**Accepts:** String or array[string]
+
+### Action
+
+Adds additional functionality to certain commands.
+
+**Property name:** `action`
+
+**Necessity:** Optional
+
+**Accepts:** String
+
+<br />
+
+___
+
+## Accepted Modifiers and Keys
+
+### Modifiers
+`Ctrl+`, `Shift+`, `Alt+`
+
+### Keys
+| Type | Keys |
+| ---- | ---- |
+| Function and Alphanumeric Keys | `f1-f24`, `a-z`, `0-9` |
+| Symbols | ``` ` ```, `-`, `=`, `[`, `]`, `\`, `;`, `'`, `,`, `.`, `/` |
+| Arrow Keys | `down`, `left`, `right`, `up`, `pagedown`, `pageup`, `pgdn`, `pgup`, `end`, `home`, `plus` |
+| Action Keys | `tab`, `enter`, `esc`, `escape`, `space`, `backspace`, `delete`, `insert` |
+| Numpad Keys | `numpad_0-numpad_9`, `numpad0-numpad9`, `numpad_add`, `numpad_plus`, `numpad_decimal`, `numpad_period`, `numpad_divide`, `numpad_minus`, `numpad_subtract`, `numpad_multiply` |
+
+<br />
+
+___
+
+## Key binding formats
+
+Keybindings can be structured in the following formats:
+
+### Commands without arguments
+
+```json
+{ "command": "commandName", "keys": "modifiers+key" }
+```
+
+### Commands with arguments
+```json
+{ "command": { "action": "commandName", "argument": "value" }, "keys": "modifiers+key" }
+```
+
+<br />
+
+___
+
+## Application-level commands
+
+### Close window
+
+Close the current window and all tabs within it.
+
+**Command name:** `closeWindow`
+
+**Default binding:**
+```json
+{ "command": "closeWindow", "keys": "alt+f4" }
+```
```

Comment:
gif? Should we mention that there's a popup dialog and it can be disabled via a setting?